### DISCUSSION TITLE
Install and run the latest js7 agent on windows server 2003 SP1

### DISCUSSION THREAD

**Discussion by Arsene-K** [sos-berlin/js7#4] [Q&A] [answered]
Hello,
Is it possible to install js7 agent on windows server 2003 SP1? And run jobs correctly ?
We've to manage automaticaly multiple tasks on our ERP and we're trying to find the best solution,

Regards

**Comment by sosap** [comment, marked as answer]
Hi,

The JS7 Agent requires Java 1.8, 11 or 17. Chances do not look good for this prerequisite:

* Oracle docs do not include Windows 2003 as a supported platform for Java, see [https://www.java.com/en/download/help/sysreq.html](https://www.java.com/en/download/help/sysreq.html)
* To my knowledge Windows Server 2003 reached its end-of-live in 2015. 

Maybe a possible alternative is to run an SSH Server on Windows 2003 and to use [JS7 Agentless Scheduling](https://kb.sos-berlin.com/display/PKB/JS7+-+Agentless+Scheduling)

**Comment by Arsene-K** [comment]
Hi,
Thanks for your answer,

You're right. But my company still use Windows Server 2003 because of our old business management software,
Notwithstanding the fact that we are in a big project of application migration, we still need to have control of our old b. m. software in read-only mode in which we are going to execute scheduled maintenance tasks with the use of an agent.

We have already tried js7 SSH Jobs using Cygwin and we got stuck on this error : 
msg=com.sos.jitl.jobs.ssh.exception.SOSJobSSHException: remote command terminated with exit code: 255

This error seems to have already been reported here : https://kb.sos-berlin.com/display/PKB/Limitations+when+using+SSH+jobs+for+Windows

Regards,

**Comment by sosap** [comment]
Hi, 

am I right to assume that you are the poster of [How to install jobscheduler js7 agent on Windows server 2003 SP1](https://stackoverflow.com/questions/72270701/how-to-install-jobscheduler-js7-agent-on-windows-server-2003-sp1/72310577#72310577) ?

Then first of all, congrats! You obviously did make it to have a matching JVM up and running on Windows 2003. As indicated by my response the warning from the installer can be ignored: it's about a symlink that is created for convenience only. The remaining question about coupling the Agent from a Controller will be resolved when we have knowledge of the errors that you find in the Controller's and Agent's logs as indicated.

**Comment by Arsene-K** [comment]
Yes this is me,

I've installed java 8_u0 (exactly 8.0.0.132). js7_agent_4445 service is running but the agent is not launched correctly,
Here are the logs
[agent-service.2022-05-18.log](https://github.com/sos-berlin/js7/files/8739981/agent-service.2022-05-18.log)
[agent-service.2022-05-20.log](https://github.com/sos-berlin/js7/files/8739983/agent-service.2022-05-20.log)
[controller.log](https://github.com/sos-berlin/js7/files/8739984/controller.log)

**Comment by sosap** [comment]
The controller.log states

> 2022-05-20T13:47:20,145 info  js7.journal.Journal -  Event  Agent:agent-vtom <-: AgentCouplingFailed(akka.stream.StreamTcpException: Tcp command [Connect(qprwva012:4445,None,List(),Some(10 seconds),true)] failed because of java.net.ConnectException: Connection refused: no further information [suppressed: TCP Connect qprwva012:4445: Connection refused: no further information])

The message is clear: The Agent cannot be reached by its host:port. Possible reasons include that 

* the firewall on the Windows 2003 server is not configured to allow incoming traffic on port 4445.
* the Agent was started for a different port or faces some problem. You will find the information from the Agent's ./logs/agent.log file (not from the agent-service.*.log files that state information about the Windows Service only.

**Comment by Arsene-K** [comment]
Thanks for your answer

I've already opened all the ports in my firewall between the controller and this machine. And I can reach it.
Here's my view : 
1. The js7 agent faces some problems although the service is launched : maybe due to the the too old version of java 8 (u0) or too old version of the OS (but you guys only can debug and tell)
2. Maybe the agent is running on a different port : But I installed it with default settings

The agent.log file remain empty that why I didn't send it.

Regards,

**Comment by sosap** [comment]
It's good to know that no agent.log file has been created.

Let me suggest that you stop the Agent's Windows Service (otherwise check that you kill the Agent's Java process from the Task Manager). Then open a command shell and navigate to the Agents ./bin directory where you should find the Agent's Instance Start Script that you can execute like this

> ./bin/agent_4445.cmd start

(running the command without argument gives you the usage clause)

This should create output to the console window, and/or to ./var_4445/logs/agent.log and/or to the watchdog.log file in the same directory.

**Comment by Arsene-K** [comment]
I got this :

![image](https://user-images.githubusercontent.com/105710265/169557339-32339337-c4a4-408c-85bd-14840a240d81.png)

Nothing in the file agent.log and no watchdog.log file

**Comment by sosap** [comment]
Please set your JAVA_HOME environment variable to your JRE, not JDK. For example 

* if Java is installed in C:\Program Files\Java\jdk8u202-b08
* then - before you run the Agent Instance Start Script - use the command: set JAVA_HOME=C:\Program Files\Java\jdk8u202-b08\jre

**Comment by Arsene-K** [comment]
Hello @sosap,
I have already set the good path for JAVA_HOME,
Unfortunately I still have the same error (and nothing is written in the agent.log file)

![image](https://user-images.githubusercontent.com/105710265/169818763-c4d3f0d2-c685-477e-8132-c5484d958024.png)

**Comment by sosap** [comment]
The agent.log file cannot be created as Java crashes and logging is performed with Log4j. This is why you see output in stdout.

The issue most probably is due to an old Java version. Oracle reports that [JDK bug 8046233]( https://www.oracle.com/java/technologies/javase/8u25-bugfixes.html) should be resolved, some users report that JRE 1.8.0.144 resolves the problem. I cannot verify this information. Best would be if you can check to use the latest Java JRE available for Windows 2003.

**Comment by Arsene-K** [comment]
I'll see what I can do

Thanks
Regards
